Source/SwiftLintFramework/Extensions/NSRegularExpression+SwiftLint.swift

```diff
@@ -9,7 +9,9 @@
 import Foundation
 
 #if os(Linux)
+#if !swift(>=3.1)
 public typealias NSRegularExpression = RegularExpression
+#endif
 public typealias NSTextCheckingResult = TextCheckingResult
```

Comment:
TextCheckingResult is gaining the NS back in a future version of linux foundation.
https://github.com/apple/swift-corelibs-foundation/pull/871